### PR TITLE
[GAIA] Support Primary Key Optimization in GraphScopeStore Adapter

### PR DESCRIPTION
**Pull request by BingqingLyu**
## Related issue number

#501

### CODE REVIEW

File: research/gaia/gremlin/gs_gremlin/src/graph_proxy/global_storage.rs [hunk 1]
```diff
@@ -451,3 +468,41 @@ fn get_partition_vertex_ids(
     let partition_id = graph_partition_manager.get_partition_id(id as VertexId) as PartitionId;
     vec![(partition_id, vec![id as VertexId])]
 }
+
+/// Get vertex ids by giving label_ids and corresponding primary key values
+fn get_vertex_ids_by_primary_keys(
```
Comment (longbinlai) [inline research/gaia/gremlin/gs_gremlin/src/graph_proxy/global_storage.rs:473]: This function is better a member of `GraphScopeStore`

File: research/gaia/gremlin/gs_gremlin/src/graph_proxy/global_storage.rs [hunk 2]
```diff
@@ -451,3 +468,41 @@ fn get_partition_vertex_ids(
...
+    labels: &Vec<LabelId>,
+    graph_partition_manager: Arc<dyn GraphPartitionManager>,
+    primary_key_list: Vec<String>,
+) -> Vec<ID> {
+    let mut ids = vec![];
+    for label_id in labels {
+        for primary_key in primary_key_list.iter() {
```
Comment (longbinlai) [inline research/gaia/gremlin/gs_gremlin/src/graph_proxy/global_storage.rs:480]: `for primary_key in &primary_key_list:`

File: research/gaia/gremlin/gs_gremlin/src/graph_proxy/global_storage.rs [hunk 3]
```diff
@@ -451,3 +468,41 @@ fn get_partition_vertex_ids(
     let partition_id = graph_partition_manager.get_partition_id(id as VertexId) as PartitionId;
     vec![(partition_id, vec![id as VertexId])]
 }
+
+/// Get vertex ids by giving label_ids and corresponding primary key values
+fn get_vertex_ids_by_primary_keys(
+    labels: &Vec<LabelId>,
+    graph_partition_manager: Arc<dyn GraphPartitionManager>,
+    primary_key_list: Vec<String>,
```
Comment (longbinlai) [inline research/gaia/gremlin/gs_gremlin/src/graph_proxy/global_storage.rs:476]: Why primary_key is a string type? property has been transformed as id no?

Comment (sighingnow) [reply research/gaia/gremlin/gs_gremlin/src/graph_proxy/global_storage.rs:476]: Different primary key types (e.g., `int64`, `string`) are handled by a same FFI function, and a string argument is expected. For int types, we first cast it to string by `to_string()`, e.g., `1` -> `"1"`.

**Pull request closed without merging**